components/patient-data-sharing/receiver-api/src/main/java/org/phenotips/data/receive/internal/DefaultReceivePatientData.java

```diff
@@ -382,44 +394,113 @@ protected boolean userTokensEnabled(BaseObject serverConfig)
         return DEFAULT_USER_TOKENS_ENABLED;
     }
 
+    // internal class which contains details about a login attempt
+    protected class LoginResult
+    {
+        private String authorizedUsername;
+        private String userToken;
+
+        private JSONObject failedLoginDetails;
+
+        public LoginResult(String authorizedUsername, String userToken) {
+            this.authorizedUsername = authorizedUsername;
+            this.userToken = userToken;
+        }
+
+        public LoginResult(JSONObject failedLoginDetails) {
+            this.failedLoginDetails = failedLoginDetails == null
+                    ? generateFailedLoginResponse()
+                    : failedLoginDetails;
+        }
+
+        public boolean isSuccessful() {
+            return this.authorizedUsername != null;
+        }
+
+        public boolean isFailed() {
+            return !this.isSuccessful();
+        }
+
+        public String getAuthorizedusername() {
```

Comment:
Please standardize this method's name:
```suggestion
        public String getAuthorizedUsername() {
```